soci/artifacts.go

```diff
@@ -418,6 +418,16 @@ func (db *ArtifactsDb) WriteArtifactEntry(entry *ArtifactEntry) error {
 	return err
 }
 
+func (db *ArtifactsDb) fixSociV2ArtifactReference(indexDigest string, manifestDigest string, imageDigest string) error {
+	ae, err := db.GetArtifactEntry(indexDigest)
+	if err != nil {
+		return err
+	}
+	ae.ImageDigest = imageDigest
+	ae.OriginalDigest = manifestDigest
+	return db.WriteArtifactEntry(ae)
```

Comment:
Do we care about handling concurrent read/writes here?

Reply:
After offline discussion: it's easy enough to put these into a transaction so we just don't have any risk.